CMake/ITKFactoryRegistration.cmake

```diff
@@ -14,7 +14,7 @@
 # Overview: Static registration of ITK factories
 # ----------------------------------------------
 #
-# For each factory type (Image, Transform, Mesh, FFT, ...), a
```

Comment:
There are two commits in this PR. I guess you did not squash them on purpose?

Reply:
The first is @thewtex 's initial work, I will leave it up to him if he would rather I squash them both into one.

Reply:
It looks like he is out for the next couple of days. @dzenanz would you rather I go ahead and merge or wait for additional reviews?

Reply:
I would rather wait. Reverting commits in master is a poor style, and this is a big one with many possible consequences.

Reply:
@tbirdso this patch has evolved into a very nice result that serves the project's needs. Thanks for your hard work. The improvements in module dependency loading were involved. Please go ahead and squash and merge!